### PR TITLE
Add new features on app

### PR DESCRIPTION
**Pull request by bh-2024**
Here are changes added on react component and also python change

### CODE REVIEW

File: Indie_Gems_Portal/scripts/sort_genres.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# File: process_genres.py
```
Comment (bh-2024) [inline Indie_Gems_Portal/scripts/sort_genres.py:1]: Based on the provided code review suggestions, here are the most important considerations:

1. **Error Handling**: Implement a broader `try-except` block to catch more general exceptions related to parsing and processing the JSON data.
2. **Data Validation**: Use a library like `jsonschema` or `pydantic` to ensure that the input data conforms to a predefined schema, catching errors early and making the script more robust.
3. **Code Organization**: Refactor the script to separate the parsing of JSON data from the sorting and filtering logic, improving readability and maintainability.

These considerations are crucial for ensuring the reliability and performance of the script, and addressing potential issues such as incomplete error handling and incorrect assumptions about input data.

File: Indie_Gems_Portal/scripts/sort_genres_test [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# File: test_sort_genres.py
```
Comment (bh-2024) [inline Indie_Gems_Portal/scripts/sort_genres_test:1]: Based on the provided code review suggestions, here are the most important considerations:

1. **Add more test cases**: The current test suite covers some but not all possible scenarios. Adding tests for edge cases like empty input data, duplicate genres, missing or invalid data, and large datasets will improve coverage and robustness.
2. **Consider using a testing framework with more features**: While `unittest` is a good starting point, frameworks like `pytest` offer more flexibility and features that can make testing easier and more efficient.
3. **Use meaningful variable names and docstrings**: Clear and descriptive variable names and docstrings will improve code readability and make it easier for others to understand the code.

By addressing these areas, you can strengthen your test suite and ensure that your `sort_genres` module functions correctly in a variety of scenarios.

File: Indie_Gems_Portal/src/components/GenreList.tsx [hunk 3]
```diff
@@ -8,6 +8,7 @@ import {
   Spinner,
   Text,
 } from "@chakra-ui/react";
+import React, {useEffect, useState} from "react";
```
Comment (bh-2024) [inline Indie_Gems_Portal/src/components/GenreList.tsx:11]: Here are the most important code review suggestions:

1. **Provide a brief description of the custom hook (`useGenres`)**: The hook is used without explanation, making it difficult for developers to understand its implementation and behavior.
2. **Implement fallback logic when data fetching fails**: No clear error handling mechanism is implemented, leaving users with no clear indication that an error occurred or how to recover from it.
3. **Use more descriptive and consistent variable names**: Names like `newData` and `data` are not very descriptive, making the code harder to read and understand.
4. **Add additional error handling mechanisms**: Consider retrying failed requests or displaying an error message to the user for a better user experience.

By addressing these suggestions, you can improve the overall quality, readability, and effectiveness of your `GenreList` component.
